frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -0,0 +1,141 @@
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { Resources } from '../import/import-types';
+
+export const getResourcesType = (resource: K8sResourceKind): string => {
+  switch (resource.kind) {
+    case 'DeploymentConfig':
+      return Resources.OpenShift;
+    case 'Deployment':
+      return Resources.Kubernetes;
+    default:
+      return Resources.KnativeService;
+  }
+};
+
+const checkIfTriggerExists = (triggers: { [key: string]: string | {} }[], type: string) => {
+  return !!_.find(triggers, (trigger) => {
+    return trigger.type === type;
+  });
+};
+
+export const getGitData = (buildConfig: K8sResourceKind) => {
+  const gitData = {
+    url: _.get(buildConfig, 'spec.source.git.uri', ''),
+    type: _.get(buildConfig, 'spec.source.type', ''),
+    ref: _.get(buildConfig, 'spec.source.git.ref', ''),
+    dir: _.get(buildConfig, 'spec.source.contextDir', ''),
+    showGitType: false,
+    secret: _.get(buildConfig, 'spec.source.sourceSecret.name', ''),
+    isUrlValidated: false,
+    isUrlValidating: false,
+  };
+  return gitData;
+};
+
+export const getRouteData = (route: K8sResourceKind, resource: K8sResourceKind) => {
+  let routeData = {
+    show: _.isEmpty(route),
+    create: true,
+    targetPort: _.get(route, 'spec.port.targetPort', ''),
+    unknownTargetPort: '',
+    path: _.get(route, 'spec.path', ''),
+    hostname: _.get(route, 'spec.host', ''),
+    secure: _.has(route, 'spec.termination'),
+    tls: {
+      termination: _.get(route, 'spec.termination', ''),
+      insecureEdgeTerminationPolicy: _.get(route, 'spec.insecureEdgeTerminationPolicy', ''),
+      caCertificate: _.get(route, 'spec.caCertificate', ''),
+      certificate: _.get(route, 'spec.certificate', ''),
+      destinationCACertificate: _.get(route, 'spec.destinationCACertificate', ''),
+      privateKey: _.get(route, 'spec.privateKey', ''),
+    },
+  };
+  if (getResourcesType(resource) === Resources.KnativeService) {
+    const port = _.get(resource, 'spec.template.spec.containers[0].ports[0].containerPort', '');
+    routeData = {
+      ...routeData,
+      show:
+        _.get(resource, 'metadata.labels["serving.knative.dev/visibility"]', '') ===
+        'cluster-local',
+      unknownTargetPort: _.toString(port),
+    };
+  }
+  return routeData;
+};
+
+export const getBuildData = (buildConfig: K8sResourceKind) => {
+  const buildStrategyType = _.get(buildConfig, 'spec.strategy.type');
+  const buildStrategyData =
+    buildStrategyType === 'Source'
+      ? _.get(buildConfig, 'spec.strategy.sourceStrategy')
+      : _.get(buildConfig, 'spec.strategy.dockerStrategy');
+  const triggers = _.get(buildConfig, 'spec.triggers');
+  const buildData = {
+    env: buildStrategyData.env,
```

Comment:
I deployed an image from the internal registry. On the Topology page, when the pod started running I right clicked on the node and selected `Edit Application` from the context menu. The modal didnt't show up and could see the below error in the browser's console. You might want to handle this scenario. Please do let me know if I am missing something

![Screenshot from 2019-12-18 12-26-48](https://user-images.githubusercontent.com/22490998/71063646-1f76a600-2193-11ea-960f-6f239f447edb.png)

Reply:
So support for apps created using deploy image is not added yet as the corresponding story is yet to be started. I will anyways add a check for now, so that it doesnt try to render anything if app is created using deploy image flow.

Reply:
yes, I have seen that in the description. It's just that I felt we should gracefully handle the scenario till we have the feature in place :) For a sec it almost looks like the modal is going to open then the error occurs

Reply:
Sure, will handle this in this PR.

Reply:
Done.